jigglypuffRL/deeprl/ddpg/ddpg.py

```diff
@@ -236,7 +235,7 @@ def learn(self):
         for t in range(total_steps):
             # execute single transition
             if t > self.start_steps:
-                action = self.select_action(state, deterministic=False)
+                action = self.select_action(state, deterministic=True)
```

Comment:
Is it training without deterministic actions? Because ideally it should be taking deterministic actions only during Testing. I know Td3 was, so just confirming

Reply:
Its not exactly taking deterministic actions now. deterministic=True just ensures that `ac.actor.get_action(s)` returns just the value at the final layer. Then in `select_action()` we're adding noise, which adds stochasticity

Reply:
Why though? Shouldnt we sample from the distribution if its not deterministic? I guess we're sort of missing passing the deterministic arg to `get_action`?

Reply:
According to the DDPG paper, I think the policy outputs a deterministic action to which we add noise (OU or normal) and then clip from -action_lim to action_lim. As of now this is what I have implemented in the DDPG select_action method. But this could be modified to instead make use of the BasePolicy get_action's non deterministic action if that's what you mean? 

But if we do that, we can't use OU noise. Only normal

Reply:
Gotcha, understand this now. I was getting confused with normal policy gradients. Forgot this has to be determinitics

Reply:
What you said will reduce the size of `select action ` but prevent us from using other kinds of noise. Hope it makes sense

Reply:
Yup